version: '3.4'

services:

  nginx:
    image: nginx:1.21.1
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - web-static:/var/www/livedata/static
      - ./config/nginx/django.conf:/etc/nginx/conf.d/django.conf
      - ./config/nginx/docker-entrypoint.sh:/docker-entrypoint.d/docker-entrypoint.sh
      - ./third-party/wait-for-it.sh:/usr/bin/wait-for-it.sh
    depends_on:
      - livedata

  db:
    image: postgres:9.6.22
    environment:
      POSTGRES_DB: ${DATABASE_NAME}
      POSTGRES_USER: ${DATABASE_USER}
      POSTGRES_PASSWORD: ${DATABASE_PASS}
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DATABASE_USER} -d ${DATABASE_NAME}"]
      interval: 5s
      timeout: 5s
      retries: 5
    ports:
      - "${DATABASE_PORT}:${DATABASE_PORT}"
    volumes:
      - db-data:/var/lib/postgresql/data/

  livedata:
    image: live_data:dev
    ports:
      - 9999:8000
    environment:
      DATABASE_NAME: ${DATABASE_NAME}
      DATABASE_USER: ${DATABASE_USER}
      DATABASE_PASS: ${DATABASE_PASS}
      DATABASE_HOST: ${DATABASE_HOST}
      DATABASE_PORT: ${DATABASE_PORT}
    depends_on:
      db:
          condition: service_healthy

volumes:
  db-data: